### DISCUSSION TITLE
Proposal for backing field to be autogenerated for all properties

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/csharplang#4287] [Language Ideas] [answered]
Right now if I need to add any logic to getter or setter of the property, I need to explicitly create backing field myself. It would be great if backing field creation would still be handled by compiler even if getter and/or setter have custom logic. 

Backing field should be generated only if getter or setter references `backing_field`, so no unused fields would be created.

### Example:
Before:
```cs
private string _hideSelectedEventsIcon = MaterialIconsFont.Filter;
public string HideSelectedEventsIcon 
{ 
    get => _hideSelectedEventsIcon; 
    set => SetProperty(ref _hideSelectedEventsIcon, value); 
}

private string _notNullString;
public string NotNullString 
{ 
    get => _notNullString ?? string.Empty;
    set => _notNullString = value; 
}
```

After:
```cs
public string HideSelectedEventsIcon 
{ 
    get;
    set => SetProperty(ref backing_field, value); 
} = MaterialIconsFont.Filter;

public string NotNullString
{ 
    get => backing_field ?? string.Empty;
    set;
}
```

(not sure what keyword could be used for to reference backing field (to get it's current value and set new one if needed), so `backing_field` is a placeholder for now)

### Benefits:
- Reduce amount of boilerplate code
- Reduce amount of code issues when value is accidentally set to backing field instead of property
- Reduce amount of code issues when wrong backing field is used inside property's getter or setter (because copy-paste, similar name, or other reason)

### Drawbacks
- I think we'll need new keyword to represent backing field that would exist only within property's getter and setter (the same way `value` only exists within setter)

**Comment by JanTamis** [comment]
I think this is a duplicate but im not sure which one.

**Reply by maxkoshevoi** [reply]
This looks like an obvious idea, so could be a duplicate. I didn't find related discussion or issue. 
Closest I could find is this: https://github.com/dotnet/csharplang/issues/2657, but that's not what I'm describing here.

**Comment by HaloFour** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/issues/140

**Reply by JanTamis** [reply]
That is the one, thanks
